subsys/bluetooth/mesh/shell/shell_onoff_cli.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <bluetooth/mesh/models.h>
+#include <shell/shell.h>
+
+#include "mesh/net.h"
+#include "mesh/access.h"
+#include "shell_model_utils.h"
+
+#include <shell/shell.h>
+#include <stdlib.h>
+
+static struct bt_mesh_model *mod;
+
+static int shell_onoff_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (!mod) {
+		mod = shell_model_first_get(BT_MESH_MODEL_ID_GEN_ONOFF_CLI);
+	}
+
+	struct bt_mesh_onoff_cli *cli = mod->user_data;
+
+	return bt_mesh_onoff_cli_get(cli, NULL, NULL);
+}
+
+static int onoff_set(size_t argc, char *argv[], bool acked)
+{
+	if (argc < 3) {
+		return -EINVAL;
+	}
+
+	bool on_off = (bool)strtol(argv[1], NULL, 0);
+	uint32_t time = (uint32_t)strtol(argv[2], NULL, 0);
+	uint32_t delay = (uint32_t)strtol(argv[3], NULL, 0);
+
+	if (!mod) {
+		mod = shell_model_first_get(BT_MESH_MODEL_ID_GEN_ONOFF_CLI);
+	}
+
+	struct bt_mesh_onoff_cli *cli = mod->user_data;
+	struct bt_mesh_model_transition trans = { .time = time, .delay = delay };
+	struct bt_mesh_onoff_set set = { .on_off = on_off, .transition = &trans };
+
+	if (acked) {
+		return bt_mesh_onoff_cli_set(cli, NULL, &set, NULL);
+	} else {
+		return bt_mesh_onoff_cli_set_unack(cli, NULL, &set);
+	}
+}
+
+static int shell_onoff_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	return onoff_set(argc, argv, true);
+}
+
+static int shell_onoff_set_unack(const struct shell *shell, size_t argc, char *argv[])
+{
+	return onoff_set(argc, argv, false);
+}
+
+static int shell_instance_set(const struct shell *shell, size_t argc, char *argv[])
+{
+	if (argc < 1) {
+		return -EINVAL;
+	}
+
+	struct bt_mesh_model *mod_temp;
+	uint16_t addr = (uint16_t)strtol(argv[1], NULL, 0);
+
+	mod_temp = shell_model_get(BT_MESH_MODEL_ID_GEN_ONOFF_CLI, addr);
+
+	if (mod_temp) {
+		mod = mod_temp;
+	} else {
+		shell_error(shell, "Unable to find model instance for element address %d", addr);
+	}
+
+	return 0;
+}
+
+static int shell_instances_get(const struct shell *shell, size_t argc, char *argv[])
+{
+	uint8_t elem_cnt = bt_mesh_elem_count();
+	uint16_t mod_arr[elem_cnt];
```

Comment:
My thought was to allocate an array that is able to hold the maximum number of model instances of a spesific type that can be found in a device. Since a single element can only support one instance of the model i thougth this made sence. `onoff_cli_model_cnt <= element_cnt`